pkg/policy/mapstate.go

```diff
@@ -241,13 +318,20 @@ func (keys MapState) denyPreferredInsertWithChanges(newKey Key, newEntry MapStat
 			for k, v := range keys {
 				if newKey.TrafficDirection == k.TrafficDirection &&
 					!v.IsDeny &&
-					k.Identity == 0 {
+					k.Identity == 0 && (k.DestPort != 0 || k.Nexthdr != 0) {
```

Comment:
What happens if there's an allow-all entry already and now we're introducing an l3-only deny?

Reply:
Haven't checked now, but as per the datapath lookup order nothing special should be needed, as the L3-only deny will be looked up before the allow-all. Deny rules always take precedence so this is the correct thing to do.

Reply:
Ah okay. At some level that makes sense. I think I was just caught up a bit by the fact that this condition was changed, so I was wondering if we were just running extra unnecessary code in this snippet before for that case or we could have limited this condition to L4-only entries from the beginning.

Like is this change by itself safe regardless of the rest of the PR? Or is it depending on some other logic being changed somewhere else?

Reply:
Added a test case to cover this explicitly (`test-4 - Adding L3-deny to an existing allow-all`)